### PR TITLE
Basic build script

### PR DESCRIPTION
**Pull request by MartinPetkov**
Will use this for releasing the first version. Working on basic release docs too.

### CODE REVIEW

**Comment by umairidris** [comment]
Should this be under build/ ?

Maybe build/release.sh ?

**Comment by MartinPetkov** [comment]
> Should this be under build/ ?
> 
> Maybe build/release.sh ?

Good idea.

**Comment by MartinPetkov** [comment]
> Should this be under build/ ?
> 
> Maybe build/release.sh ?

I wouldn't call it release.sh though. I think that's too vague and people may assume it's actually creating tags or GitHub releases too, when it's only building artifacts.

File: build/build-release.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+fi
+
+version_regex="^v[0-9]+\.[0-9]+\.[0-9]+$"
+if ! [[ ${VERSION} =~ ${version_regex} ]]; then
+  echo "Version \"${VERSION}\" does not match regex ${version_regex}"
+  exit 1
+fi
+
+# Build binaries
+SUPPORTED_OS=("linux" "darwin" "windows")
```
Comment (xingao267) [inline build/build-release.sh:49]: Just linux and darwin? since we didn't verify windows

Comment (MartinPetkov) [reply build/build-release.sh:49]: I just tested it on Windows (using `scripts/regen.sh` but converted to run in Windows `cmd`). Seems to work as expected.

**Pull request merged**